diff --git a/libgambatte/libretro/libretro.cpp b/libgambatte/libretro/libretro.cpp
index 74b2b1d..caa2d1e 100644
--- a/libgambatte/libretro/libretro.cpp
+++ b/libgambatte/libretro/libretro.cpp
@@ -2387,6 +2387,12 @@ static void check_variables(bool startup)
          {
             rgb32 = gb.gbcToRgb32(gbc_bios_palette[palnum * 4 + colornum]);
             gb.setDmgPaletteColor(palnum, colornum, rgb32);
+
+			if (palnum==0 && colornum==0) {
+			    char cmd[32];
+			    sprintf(cmd, "echo %i > /tmp/dmg_grid_color", rgb32);
+			    system(cmd);
+			}
          }
       }
    }
